src/classes/GRER_SyncGradeLevel_TDTM.cls

```diff
@@ -83,20 +117,61 @@ global with sharing class GRER_SyncGradeLevel_TDTM extends hed.TDTM_Runnable {
                                                     FROM Grade_Enrollments__r
                                                     WHERE Status__c = 'Current') 
                                             FROM Contact
-                                            WHERE Id in :relatedContactIds]) {
-                for (Grade_Enrollment__c gradeErol : relatedContact.Grade_Enrollments__r) {
-                    if (currentGradeEnrolIds.contains(gradeErol.Id)) {
-                        relatedContact.Current_Grade_Level__c = gradeErol.Grade_Level__c;
+                                            WHERE Id IN :relatedContactIds]) {
+                for (Grade_Enrollment__c gradeEnrol : relatedContact.Grade_Enrollments__r) {
+                    if (currentGradeEnrolIds.contains(gradeEnrol.Id)) {
+                        relatedContact.Current_Grade_Level__c = gradeEnrol.Grade_Level__c;
                         contactsToUpdate.add(relatedContact);
                     } else {
-                        gradeErol.Status__c = 'Former';
-                        gradeEnrolsToUpdate.add(gradeErol);    
+                        gradeEnrol.Status__c = 'Former';
+                        gradeEnrolsToUpdate.add(gradeEnrol);    
                     }
                 }
-            }  
+            }
+            /* Query and loop through related contacts and their Current Grade Enrollments
+             * If there are no Current Grade Enrollments then mark the Current Grade Level on the Contact as 'Not Applicable'
+             * If there is only 1 Current Grade Enrollment then update the related Contact's Current Grade level to match Grade Enrollment's Current Grade level
+             * If there are more than 1 Current Grade Enrollments then show an error on the records that have Grade levels matching with Contact's Current Grade Level
+            */
+            if (contactsIdsFormer != NULL && contactsIdsFormer.size() > 0) {
+                for (Contact relatedContact :   [SELECT Id,
```

Comment:
Any chance if we can avoid querying again? We can still build the formerContactIds set and check if it contains the GR.Contact__c? Just a thought.